chat-plugins/jeopardy.js

```diff
@@ -1,772 +1,725 @@
-			data.push(this.grid[category][q]);
-		}
-		return data;
-	};
-	JeopardyQuestions.prototype.import = function (category, start, end, data) {
-		let q1 = start;
-		let q2 = 0;
-		for (; q1 < end && typeof data[q2] === 'object'; ++q1, ++q2) {
-			if (typeof data[q2].value === 'string') this.grid[category][q1].value = data[q2].value;
-			if (typeof data[q2].answer === 'string') this.grid[category][q1].answer = data[q2].answer;
-			this.grid[category][q1].isDailyDouble = !!data[q2].isDailyDouble;
+class Jeopardy extends Rooms.RoomGame {
+	constructor(room, user, categoryCount, questionCount) {
+		super(room);
+		if (room.gameNumber) {
+			room.gameNumber++;
+		} else {
+			room.gameNumber = 1;
 		}
-		return q1 - start;
-	};
-
-	JeopardyQuestions.prototype.getCategory = function (category) {
-		return this.categories[category];
-	};
-	JeopardyQuestions.prototype.setCategory = function (category, value) {
-		this.categories[category] = value;
-		this.save();
-	};
-
-	JeopardyQuestions.prototype.getQuestion = function (category, question) {
-		return {
-			value: this.grid[category][question].value,
-			answer: this.grid[category][question].answer,
-			points: calculatePoints(category, question),
-			isDailyDouble: this.grid[category][question].isDailyDouble,
-			isRevealed: this.revealedGrid[category][question],
-		};
-	};
-	JeopardyQuestions.prototype.setQuestion = function (category, question, value) {
-		this.grid[category][question].value = value;
-		this.save();
-	};
-	JeopardyQuestions.prototype.setAnswer = function (category, question, value) {
-		this.grid[category][question].answer = value;
-		this.save();
-	};
-	JeopardyQuestions.prototype.setDailyDouble = function (category, question, isDailyDouble) {
-		this.grid[category][question].isDailyDouble = isDailyDouble;
-		this.save();
-	};
-
-	JeopardyQuestions.prototype.setRevealed = function (category, question, isRevealed) {
-		this.revealedGrid[category][question] = isRevealed;
-	};
-
-	return JeopardyQuestions;
-})();
-
-let Jeopardy = (() => {
-	function Jeopardy(host, room, categoryCount, questionCount) {
-		this.host = host;
-		this.room = room;
+		this.host = user;
+		this.allowRenames = true;
+		this.state = "signups";
+		this.gameid = 'jeopardy';
+		this.title = 'Jeopardy';
+		this.questions = [];
+		this.started = false;
+		this.buzzed = new Set();
 		this.categoryCount = categoryCount;
 		this.questionCount = questionCount;
-
-		this.users = new Map();
-		this.questions = new JeopardyQuestions(room, categoryCount, questionCount);
-
-		this.isStarted = false;
-		this.remainingQuestions = categoryCount * questionCount;
-
-		this.currentUser = null;
-		this.currentCategory = -1;
-		this.currentQuestion = -1;
-		this.currentAnswerer = null;
-		this.currentAnswer = "";
-
-		this.remainingFinalWagers = 0;
-		this.remainingFinalAnswers = 0;
-
-		this.room.add("A new Jeopardy match has been created by " + host.name);
+		this.round = 1;
+		this.points = new Map();
```

Comment:
You're allowing renames in this.  You probably should make a separate class for the player, and put move ``this.points``, ``this.wagers``, ``this.finalanswers``, ``this.answers``, ``this.buzzedEarly`` to the player object where they will be affected by ``RoomGame#onRename()``

Stylistical thing, but __probably__ making all the variables camel case or all lowercase would be better.... but not my call